### PR TITLE
Performance: remove unwanted object instance in BasicInvocationFuture.

### LINKED ISSUE #4407: Optimization: No need for IsStillExecutingOperation for local calls
There is no need to use an IsStillExecutingOperation to check if an operation is running locally. One can directly access the scheduler.

### LINKED ISSUE #4408: Optimization: BasicInvocationFuture.waiterCount replace AtomicInteger by AtomicIntegerFieldUpdater
No need to create an additional AtomicInteger object; one can replace it by an AtomicIntegerFieldUpdater like already is done for the response field in the same class.

### PR DESCRIPTION
**Pull request by pveentjer**
Fixes #4408 

Removal of unwanted AtomicInteger in BasicInvocationFuture. It is replaced by an AtomicIntegerFieldUpdater.

### CODE REVIEW

**Comment by gurbuzali** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/spi/impl/BasicInvocationFuture.java [hunk 1]
```diff
@@ -283,7 +287,7 @@ private static long getCallTimeoutExtension(long callTimeout) {
     }
 
     int getWaitingThreadsCount() {
-        return waiterCount.get();
+        return WAITER_COUNT_FIELD_UPDATER.get(this);
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/BasicInvocationFuture.java]: I could even simplify that with a return waiterCount; no need to go through the field updater.

**Comment by pveentjer** [comment]
@gurbuzali @jerrinot can you have (another) look at this one?

**Comment by gurbuzali** [comment]
looks good

**Comment by pveentjer** [comment]
Can someone merge this one?

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**